engine/engine.go

```diff
@@ -52,19 +53,22 @@ func (e *Engine) Setup(ctx context.Context, pipelineConfig *spec.PipelineConfig)
 	}
 	// create volumes
 	for _, vol := range pipelineConfig.Volumes {
-		if vol != nil && vol.HostPath != nil {
-			path := vol.HostPath.Path
-			vol.HostPath.Path = pathConverter(path)
+		if vol == nil || vol.HostPath == nil {
+			continue
+		}
+		path := vol.HostPath.Path
+		vol.HostPath.Path = pathConverter(path)
 
-			if _, err := os.Stat(path); err == nil {
-				continue
-			}
+		if _, err := os.Stat(path); err == nil {
+			_ = os.Chmod(path, permissions)
+			continue
+		}
 
-			if err := os.MkdirAll(path, 0777); err != nil { //nolint:gomnd
-				return errors.Wrap(err,
-					fmt.Sprintf("failed to create directory for host volume path: %q", path))
-			}
+		if err := os.MkdirAll(path, permissions); err != nil {
+			return errors.Wrap(err,
+				fmt.Sprintf("failed to create directory for host volume path: %q", path))
 		}
+		_ = os.Chmod(path, permissions)
```

Comment:
Why is this needed?

Reply:
We still need to give it 0777 if it was created right now